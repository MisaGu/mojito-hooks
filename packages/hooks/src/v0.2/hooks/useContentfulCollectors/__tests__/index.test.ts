import { act, renderHook } from '@testing-library/react-hooks';
import { TestWrapper } from '../../../components/test/wrapper/TestWrapper';
import {
  useContentfulCollectors,
  UseContentfulCollectorsProps,
  UseContentfulCollectorsReturn,
} from '../useContentfulCollectors';

describe('useContentfulCollectors()', () => {
  it('should be defined', () => {
    expect(useContentfulCollectors).toBeDefined();
  });

  it('returns the right result when needed', async () => {
    const { result, rerender, waitFor } = renderHook<
      [UseContentfulCollectorsProps?],
      UseContentfulCollectorsReturn
    >(
      (args = []) => {
        return useContentfulCollectors(...args);
      },
      {
        wrapper: TestWrapper as any,
      },
    );

    // Trigger request...
    await act(async () => rerender());

    // ...and wait for response:
    await waitFor(() => result.current.queryResult.isSuccess);

    // Check data matches snapshot:
    expect(result.current).toMatchSnapshot({
      queryResult: {
        dataUpdatedAt: expect.any(Number),
      },
    });
  });
});
