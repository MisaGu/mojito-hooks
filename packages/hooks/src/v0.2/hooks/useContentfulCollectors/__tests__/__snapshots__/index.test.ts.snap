// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useContentfulCollectors() returns the right result when needed 1`] = `
Object {
  "collectors": Array [
    Object {
      "about": "...",
      "avatar": Object {
        "title": "Mock Collector 1",
        "url": "https://mock-cdn/mock-collector-1-avatar.png",
      },
      "linkedFrom": Object {
        "lotCollection": Object {
          "items": Array [
            Object {
              "mojitoId": "mock-item-id-1",
              "slug": "mock-item-1",
              "title": "Mock Item 1",
            },
            Object {
              "mojitoId": "mock-item-id-2",
              "slug": "mock-item-2",
              "title": "Mock Item 2",
            },
            Object {
              "mojitoId": "mock-item-id-3",
              "slug": "mock-item-3",
              "title": "Mock Item 3",
            },
          ],
        },
      },
      "name": "Mock Collector 1",
      "slug": "mock-collector-1",
      "smartContractAddress": "0x5a4e9865ae4c65a46a5ec69a4e65ac465e4a6ec4",
      "sys": Object {
        "publishedAt": "2021-09-29T10:07:59.396Z",
      },
      "twitterLink": "https://twitter.com/mock-collector-1",
      "videoId": null,
    },
    Object {
      "about": "2",
      "avatar": Object {
        "title": "Mock Collector 2",
        "url": "https://mock-cdn/mock-collector-2-avatar.png",
      },
      "linkedFrom": Object {
        "lotCollection": Object {
          "items": Array [
            Object {
              "mojitoId": "mock-item-id-4",
              "slug": "mock-item-4",
              "title": "Mock Item 4",
            },
            Object {
              "mojitoId": "mock-item-id-5",
              "slug": "mock-item-5",
              "title": "Mock Item 5",
            },
            Object {
              "mojitoId": "mock-item-id-6",
              "slug": "mock-item-6",
              "title": "Mock Item 6",
            },
          ],
        },
      },
      "name": "Mock Collector 2",
      "slug": "mock-collector-2",
      "smartContractAddress": "0xa4e956dc4b4cd554cd5b854a54a45ac465e4a6ec",
      "sys": Object {
        "publishedAt": "2021-09-30T04:19:36.121Z",
      },
      "twitterLink": "https://twitter.com/mock-collector-2",
      "videoId": null,
    },
  ],
  "error": null,
  "isLoading": false,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": true,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
